### DISCUSSION TITLE
Migrate conductor JAXB dependency from javax. to jakarta

### DISCUSSION THREAD

**Discussion by taoj-action** [Netflix/conductor#2495] [Q&A] [answered]
The conductor has JAXB dependency on javax.* which should be migrated to jakarta.* due to [some legal issues](https://eclipse-foundation.blog/2019/05/03/jakarta-ee-java-trademarks/).
https://github.com/Netflix/conductor/blob/main/core/build.gradle#L44
```
    // JAXB is not bundled with Java 11, dependencies added explicitly
    // These are needed by Apache BVAL
    implementation "javax.xml.bind:jaxb-api:${revJAXB}"
    implementation "javax.activation:activation:${revActivation}"
    implementation "org.glassfish.jaxb:jaxb-runtime:${revJAXB}"
```

Currently, we use conductor as a library which clashes with our internal libraries. We would like to make the following changes:
```
    implementation "jakarta.xml.bind:jakarta.xml.bind-api:${revJAXB}"
    implementation "jakarta.activation:jakarta.activation-api:${revActivation}"

    // Only add it as a test dependency. The actual jaxb runtime provider is provided when building the server.
    testImplementation "org.glassfish.jaxb:jaxb-runtime:${revJAXB}"
``` 

Please advise the proposed change and we can own this change. Thanks!

**Comment by taoj-action** [comment, marked as answer]
Attached git patch for the latest main branch for reference. [jakarta.txt](https://github.com/Netflix/conductor/files/7255922/jakarta.txt)

**Reply by taoj-action** [reply]
https://github.com/Netflix/conductor/pull/2497 has implemented the change.

**Comment by buzz08** [comment]
This is really helpful, thanks @taojwmware
